src/chttpd_auth_cache.erl

```diff
@@ -95,7 +95,8 @@ get_from_cache(UserName) ->
 %% gen_server callbacks
 
 init([]) ->
-    {ok, #state{changes_pid = spawn_changes(0)}}.
+    self() ! {start_listener, 0},
```

Comment:
Frankly I would choose `#2` in this case as well but I would have to refactor the code a little bit. This would slow down the review and would require writing a test suite.